### PR TITLE
useLastLocation Tests

### LINKED ISSUE #2: useLastLocation tests

### PR DESCRIPTION
**Pull request by nobrayner**
Fixes #2

### CODE REVIEW

File: src/useLastLocation.spec.tsx [hunk 1]
```diff
@@ -1,137 +1,68 @@
+/* eslint-disable import/no-extraneous-dependencies */
 import * as React from 'react';
-import { mount, ReactWrapper } from 'enzyme';
-import { MemoryRouter, Route, Link } from 'react-router-dom';
+import { renderHook as rtlRenderHook, act } from '@testing-library/react-hooks';
+import { Router } from 'react-router-dom';
```
Comment (mpeyper) [inline src/useLastLocation.spec.tsx:4]: Given the test is using `react-router-dom`, I wonder if it's better to use `@testing-library/react-hooks/dom` and remove any ambiguity about which renderer and what sort of environments is being used?

File: src/useLastLocation.spec.tsx [hunk 2]
```diff
@@ -1,137 +1,68 @@
...
 
-    $.click($.aboutLink);
-    expect($.hasLastLocation).toBe(true);
-    expect($.lastLocationPathname).toBe('/');
+    act(() => {
+      history.push('/about');
+    });
+
+    expect(result.current).not.toBe(null);
+    // @ts-ignore
```
Comment (mpeyper) [inline src/useLastLocation.spec.tsx:45]: why does this need to be ignored?

File: src/useLastLocation.spec.tsx [hunk 3]
```diff
@@ -1,137 +1,68 @@
...
-    const $ = suite();
+    const { history, result } = renderHook();
+
+    act(() => {
+      history.push('/about');
+    });
+
+    expect(result.current).not.toBe(null);
+    // @ts-ignore
+    expect(result.current.pathname).toBe('/');
```
Comment (mpeyper) [inline src/useLastLocation.spec.tsx:58]: I know the old test did this, but IMO, this is just repeating the above test and could be removed from this test.

File: src/useLastLocation.spec.tsx [hunk 4]
```diff
@@ -1,137 +1,68 @@
+/* eslint-disable import/no-extraneous-dependencies */
```
Comment (mpeyper) [inline src/useLastLocation.spec.tsx:1]: which dependencies are extraneous?

**Comment by JacobMGEvans** [comment]
This is looking really good. Definitely a great "route" to go lol

File: src/useLastLocation.spec.tsx (hunk 4 above)
Comment (JacobMGEvans) [reply src/useLastLocation.spec.tsx:1]: TS was throwing some weird issue with where the packages are being kept

File: src/useLastLocation.spec.tsx (hunk 1 above)
Comment (JacobMGEvans) [reply src/useLastLocation.spec.tsx:4]: I agree.

**Review by JacobMGEvans — CHANGES_REQUESTED** [review]

File: src/useLastLocation.spec.tsx (hunk 4 above)
Comment (nobrayner) [reply src/useLastLocation.spec.tsx:1]: Yeah, if we can fix the TS stuff, that would be great. Didn't want to worry about it for now though...

File: src/useLastLocation.spec.tsx (hunk 2 above)
Comment (nobrayner) [reply src/useLastLocation.spec.tsx:45]: VS Code was kind-of contradicting itself with "result.current could be null". But then it wouldn't let me optional chain it (i.e. `result.current?.pathname`)

File: src/useLastLocation.spec.tsx (hunk 3 above)
Comment (nobrayner) [reply src/useLastLocation.spec.tsx:58]: Fair call

**Review by JacobMGEvans — APPROVED** [review]

**Pull request merged**